internal/controllers/synthesis/lifecycle.go

```diff
@@ -197,6 +166,41 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, nil
 }
 
+// safeToCreatePod returns true when it is safe to create a new synthesizer pod.
+func (c *podLifecycleController) safeToCreatePod(ctx context.Context, pods *corev1.PodList, comp *apiv1.Composition) (bool, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	sort.Slice(pods.Items, func(i, j int) bool { return pods.Items[i].CreationTimestamp.Before(&pods.Items[j].CreationTimestamp) })
+
+	var active bool
+	var deleting bool
+	for _, pod := range pods.Items {
+		if pod.DeletionTimestamp != nil {
+			if !deleting {
+				deleting = true
+				continue
+			}
+			return false, nil // don't create new pod if two are deleting
```

Comment:
Mind expanding on why we adopt this heuristic?

Reply:
Will try to refactor a bit for clarity here